add minimal declarative config doc (#1102)
elastic/elastic-otel-java: docs/reference/edot-java/migration.md

The `OTEL_JAVA_EXPERIMENTAL_SPAN_STACKTRACE_MIN_DURATION` setting controls when stack traces are captured, but it does not provide package-based filtering.

### `transaction_ignore_urls` and `ignore_message_queues`

> With EDOT Java, [declarative configuration](./configuration.md#declarative-configuration) is supported and provides the ability to filter spans based on attributes.

Reviewer: ```suggestion
To filter spans based on attributes with EDOT Java, you'll need to use [declarative configuration](./configuration.md#declarative-configuration).
```
